modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -0,0 +1,635 @@
+    // not relevant
+    return false;
+
+  //
+  // Ok, we've established that either the element or neighbor is active on our subdomains and
+  // neither of them are remote elements, so this face is still in the running to be considered
+  // relevant. There is one more caveat to be considered. In the case that we are a boundary face,
+  // we will generally need a two term expansion to compute our value, which will require a
+  // cell-gradient evaluation. If that is the case, then all of our surrounding neighbors cannot be
+  // remote. If we are not a boundary face, then at this point we're safe
+  //
+
+  if (fi.neighborPtr())
+    return true;
+
+  mooseAssert(fi.elem().on_boundary(),
+              "If we don't have a neighbor pointer, then this element has to be on a boundary");
+
+  for (auto * const neighbor : fi.elem().neighbor_ptr_range())
+  {
+    if (!neighbor)
+      continue;
+
+    if ((neighbor == libMesh::remote_elem))
+      return false;
+  }
+
+  // We made it through all the tests!
+  return true;
+}
+
+void
+INSFVRhieChowInterpolator::insfvSetup()
+{
+  _elem_range =
+      std::make_unique<ConstElemRange>(_mesh.active_local_subdomain_set_elements_begin(_sub_ids),
+                                       _mesh.active_local_subdomain_set_elements_end(_sub_ids));
+
+  const auto & all_fi = _moose_mesh.allFaceInfo();
+  _evaluable_fi.reserve(all_fi.size());
+
+  const auto & eq = UserObject::_subproblem.es();
+  std::vector<const DofMap *> dof_maps(eq.n_systems());
+  for (const auto i : make_range(eq.n_systems()))
+  {
+    const auto & sys = eq.get_system(i);
+    dof_maps[i] = &sys.get_dof_map();
+  }
+
+  auto is_fi_evaluable = [this, &dof_maps](const FaceInfo & fi) {
+    if (!isFaceGeometricallyRelevant(fi))
+      return false;
+
+    auto is_elem_evaluable = [&dof_maps](const Elem & elem) {
+      for (const auto * const dof_map : dof_maps)
+        if (!dof_map->is_evaluable(elem))
+          return false;
+
+      return true;
+    };
+
+    // We definitely need the face info element to be evaluable in all cases
+    if (!is_elem_evaluable(fi.elem()))
+      return false;
+
+    if (fi.neighborPtr())
+      // We're on an internal face and interpolation to this face will just entail linear
+      // interpolation from neighboring cell values. We'll just check then if the neighbor is
+      // evaluable (we've already checked the element)
+      return is_elem_evaluable(fi.neighbor());
+
+    // Else we are a boundary face. Two-term boundary face extrapolation will require a cell value
+    // and gradient, which will require evaluations on all surrounding elements
+
+    mooseAssert(fi.elem().on_boundary(),
+                "If we don't have a neighbor pointer, then this element has to be on a boundary");
+
+    for (auto * const neighbor : fi.elem().neighbor_ptr_range())
+    {
+      if (!neighbor)
+        continue;
+
+      if (!is_elem_evaluable(*neighbor))
```

Comment:
No. Asking a `DofMap` whether an element is evaluable essentially asks whether for all dof indices associated with an element we can read the vector entries. So even if a variable set changes over an internal boundary, as long as the new set can be evaluated on the neighboring element, then the neighboring element would be considered evaluable.

Reply:
Strictly speaking, if there are no variables defined on the element, then it would register as evaluable. But in the geometric check we check to see whether a given element's subdomain corresponds to the blocks that *this* (INSFVRhieChowInterpolator) object is defined on ... so if there were no variables on an element that would fail in the geometric check

Reply:
ok I'm satisfied that this case is caught. 

Why are we checking all the dof maps, rather than just the ones we care about for this object (eg just the momentum variable ones, which hopefully should match, I need to make a passive aux-variable advection test ugh)

Reply:
asking about this code:
```
  const auto & eq = UserObject::_subproblem.es();
  std::vector<const DofMap *> dof_maps(eq.n_systems());
  for (const auto i : make_range(eq.n_systems()))
  {
    const auto & sys = eq.get_system(i);
    dof_maps[i] = &sys.get_dof_map();
  }

    auto is_elem_evaluable = [&dof_maps](const Elem & elem) {
      for (const auto * const dof_map : dof_maps)
        if (!dof_map->is_evaluable(elem))
          return false;

      return true;
    };
```

Reply:
I guess it's the system dof_map, not the variable one. Is there a more restricted one?

Reply:
I guess that I assume that the body force kernels could be functions of aux variables. The mixing length kernel is a function of an aux variable, isn't it? That's not a body force kernel, but I think I was trying to get out ahead of any potential use cases.

Reply:
I use `FEProblemBase::getEvaluableElementRange` in the interpolator quite a bit and that considers both the nl and aux dof maps (it actually is not as general as what I do here ... and so I'm going to update that).

This determination of the evaluable face info only will happen once for most simulations (when there is no mesh adaptivity) so I'm not too worried about cost here. Are you worried about being too restrictive in what we consider evaluable?